src/cogs/snipe.py

```diff
@@ -37,6 +40,9 @@ async def snipe(self, ctx: Context, number: int = 0):
                                   "with index 0 being the most recently deleted message")
 
         history = deleted_msgs[-1 - index]
+        if history.states[0].author.id in self._recently_banned:
+            return await ctx.send("This message could not be sniped because it's original author was banned.")
```

Comment:
make this ephemeral and delete the "++snipe" message if this was executed as text command